File: CarConfigurator/ClientApp/src/app/models/CarModelOptionProduct.ts
export class CarModelOptionProduct {
  private _name: string;
  private _description: string;

  public get name(): string {
    return this._name;
  }
  public get description(): string {
    return this._description;
  }

  constructor(productId: string, name: string, description: string, optionValues: CarModelOptionProduct[]) {
    this._name = name;
    this._description = description;
  }
}

File: CarConfigurator.BL/Models/CarModelOptionProduct.cs
namespace CarConfigurator.BL.Models
{
    public class CarModelOptionProduct : IProduct
    {
        public string EAN { get; }
        public string Name { get; }
        public string Description { get; }
        public decimal Price { get; }
        public bool IsDefault { get; }

        public CarModelOptionProduct(string ean, string name, string description, decimal price, bool isDefault)
        {
            EAN = ean;
            Name = name;
            Description = description;
            Price = price;
            IsDefault = isDefault;
        }
    }
}